src/control/server/ctl_storage.go

```diff
@@ -158,68 +157,46 @@ func (c *StorageControlService) PrepareNvme(req PrepareNvmeRequest) error {
 	)
 }
 
-type PrepareScmRequest struct {
-	Reset bool
-}
-
 // GetScmState performs required initialisation and returns current state
 // of SCM module preparation.
 func (c *StorageControlService) GetScmState() (types.ScmState, error) {
 	state := types.ScmStateUnknown
 
-	ok, _ := c.scm.ext.checkSudo()
+	ok, _ := c.ext.checkSudo()
 	if !ok {
 		return state, errors.Errorf("%s must be run as root or sudo", os.Args[0])
 	}
 
-	if err := c.scm.Setup(); err != nil {
-		return state, errors.WithMessage(err, "SCM setup")
-	}
-
-	if !c.scm.initialized {
-		return state, errors.New(msgScmNotInited)
+	if _, err := c.scm.Scan(scm.ScanRequest{Rescan: true}); err != nil {
+		return state, errors.WithMessage(err, "SCM Scan")
 	}
 
-	if len(c.scm.modules) == 0 {
-		return state, errors.New(msgScmNoModules)
-	}
-
-	return c.scm.provider.GetState()
+	return c.scm.GetState()
 }
 
-// PrepareScm preps locally attached modules and returns need to reboot message,
+// ScmPrepare preps locally attached modules and returns need to reboot message,
 // list of pmem device files and error directly.
 //
 // Suitable for commands invoked directly on server, not over gRPC.
-func (c *StorageControlService) PrepareScm(req PrepareScmRequest) (needsReboot bool, pmemDevs []scm.Namespace, err error) {
-	if req.Reset {
-		// run reset to remove namespaces and clear regions
-		needsReboot, err = c.scm.PrepReset()
-		return
-	}
-
+func (c *StorageControlService) ScmPrepare(req scm.PrepareRequest) (*scm.PrepareResponse, error) {
 	// transition to the next state in SCM preparation
-	return c.scm.Prep()
+	return c.scm.Prepare(req)
 }
 
 // ScanNvme scans locally attached SSDs and returns list directly.
 //
 // Suitable for commands invoked directly on server, not over gRPC.
-func (c *StorageControlService) ScanNvme() (types.NvmeControllers, error) {
+func (c *StorageControlService) NvmeScan() (types.NvmeControllers, error) {
 	if err := c.nvme.Discover(); err != nil {
 		return nil, errors.Wrap(err, "NVMe storage scan")
 	}
 
 	return c.nvme.controllers, nil
 }
 
-// ScanScm scans locally attached modules and returns list directly.
+// ScanScm scans locally attached modules, namespaces and state of DCPM config.
 //
 // Suitable for commands invoked directly on server, not over gRPC.
-func (c *StorageControlService) ScanScm() (types.ScmModules, types.PmemDevices, error) {
-	if err := c.scm.Discover(); err != nil {
-		return nil, nil, errors.Wrap(err, "SCM storage scan")
-	}
-
-	return c.scm.modules, c.scm.pmemDevs, nil
+func (c *StorageControlService) ScmScan() (*scm.ScanResponse, error) {
+	return c.scm.Scan(scm.ScanRequest{Rescan: true})
```

Comment:
Also here. Again, if this is deliberate and intentional that's fine, but I'm concerned that it's going to add unnecessary delay in responding to client RPCs. If there's a reason that we should always rescan rather than caching, then I'd argue for just removing the caching behavior.

Reply:
yes I think we should remove the caching behaviour but maybe not part of this Pr? happy to do it here if you want

Reply:
removed the rescan